File: src/main/java/se/chalmers/cs/pgf/parse/pmcfg/TreeBuilder.java
package se.chalmers.cs.pgf.parse.pmcfg;

import se.chalmers.cs.pgf.abssyn.*;

import java.util.HashSet;
import java.util.Set;

/** This is the class responsible for building an abstract syntax tree from the chart.
 * 
 * @author gdetrez
 *
 */
public class TreeBuilder {
	static public Set<ASTree> buildTrees(Chart chart, Category startCat)
	{
		Set<ASTree> result = new HashSet<ASTree>();
		int n = chart.getSize();
		for( PassiveEdge e : chart.getPassive(0, n))
		{
			
		}
		return result;
	}
}

File: code/src/tree-builder.scala
package se.chalmers.cs.pgf.parse.pmcfg

import se.chalmers.cs.pgf.parse.trees._

class Chart

object TreeBuilder {
  def buidTrees( c : Chart ):List[Tree] = {
    for(c.getPassive(0,fin).fileter(cat = deridedCat).mkTree)
  }
  
  def mkTreeForEdge( c : Chart, e : PassiveEdge) = {
    val c = e.getFreshCat()
    forall(p <- c.getProdByReturnCat(c))
      p.
  }
  def mkTreeForCat(c : Category) = {
    forall(p <- c.getProdByReturnCat(c))
      for (c1 <- p.domain)
	c1' <- bild tree for c1
    return build tree for function p.abstractFunction with arguments the c1's
  }
    
}
